### PR TITLE
Medium application

### PR DESCRIPTION
**Pull request by sivecanta**
image
A trying to repeat Medium application

### CODE REVIEW

**Comment by sivecanta** [comment]
**Widget tree:**

https://user-images.githubusercontent.com/55202411/136978983-7f1bf37a-0a3f-4acb-aab4-b88a3f7e6cb6.mp4

**Review by max4nik — APPROVED** [review]
All looks good! Nice job!

**Comment by sivecanta** [comment]
**Here is an example of my application**:
image
image
image
image
image

**Comment by sivecanta** [comment]
**Landscape:**
image
image
image
image

**Review by maksymbudzin — APPROVED** [review]
Everything seems cool to me.I think you can merge now)

File: lib/components/article_suggestions.dart [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                child: StyledText(title, 18.0),
+              ),
+              Row(
+                children: [
+                  StyledText(pubDate, 15.0),
+                  SizedBox(width: MediaQuery.of(context).size.width * 0.03),
+                  StyledText(readingTime, 15.0),
+                  SizedBox(width: MediaQuery.of(context).size.width * 0.3),
+                  const Icon(Icons.bookmark_add_outlined,
+                      color: archiveBtnColor)
```
Comment (redd4ford) [inline lib/components/article_suggestions.dart:56]: I would consider making it a separate widget (something like `_articleController` idk, I'm not very good with naming) as you can't tell what it is unless you look at the screenshots.

File: lib/components/topic_suggestions.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  Widget _categoryTopic(BuildContext context, String title) {
+    return Container(
+        margin: const EdgeInsets.only(left: 10),
+        decoration: const BoxDecoration(
+            color: categoryCardColor,
+            borderRadius: BorderRadius.all(Radius.circular(10.0))
+        ),
+        padding: const EdgeInsets.symmetric(vertical: 5.0, horizontal: 5.0),
```
Comment (redd4ford) [inline lib/components/topic_suggestions.dart:36]: Might want to change it to `EdgeInsets.all(5.0)` :)

File: lib/components/topic_suggestions.dart [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        // alignment: ,
```
Comment (redd4ford) [inline lib/components/topic_suggestions.dart:37]: Obsolete comment, I guess?

File: lib/data.dart [hunk 4]
```diff
@@ -0,0 +1,66 @@
+class Category {
```
Comment (redd4ford) [inline lib/data.dart:1]: This whole data file is a very cool concept, just wanted to emphasize that I'm going to steal it for my next laboratory works, thank you very much

File: lib/components/navigation_bar.dart [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            ? MediaQuery.of(context).size.height * 0.07
+            : MediaQuery.of(context).size.height * 0.1,
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceAround,
+            children: const [
+              Icon(Icons.home_rounded, color: navigationActiveColor),
+              Icon(Icons.search, color: navigationNonactiveColor),
+              Icon(Icons.turned_in_not_rounded, color: navigationNonactiveColor),
+              Icon(Icons.perm_identity_rounded, color: navigationNonactiveColor)
+              // Icon()
```
Comment (redd4ford) [inline lib/components/navigation_bar.dart:23]: An obsolete comment. Might want to remove it :)

**Review by redd4ford — APPROVED** [review]
LGTM! No changes requested, just a few friendly reminders about obsolete comments and some suggestions :) Despite that, feel free to merge!

File: lib/pages/home.dart [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+class HomeScreen extends StatelessWidget {
+  const HomeScreen({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+            child: SingleChildScrollView(
+              scrollDirection: Axis.vertical,
```
Comment (Denys-Doskochynskyi) [inline lib/pages/home.dart:18]: We didn't need to write this, because it's default value

File: lib/components/navigation_bar.dart [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    return BottomAppBar(
+      color: navigationBarColor,
+      child: SizedBox(
+        height: (MediaQuery.of(context).orientation == Orientation.portrait)
+            ? MediaQuery.of(context).size.height * 0.07
+            : MediaQuery.of(context).size.height * 0.1,
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceAround,
+            children: const [
+              Icon(Icons.home_rounded, color: navigationActiveColor),
```
Comment (Denys-Doskochynskyi) [inline lib/components/navigation_bar.dart:19]: You can use a map for this, but it's also ok

File: lib/components/community_suggestions.dart [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        margin: const EdgeInsets.only(left: 10),
+        padding: const EdgeInsets.symmetric(horizontal: 5.0),
+        child: Column(
+
+          children: [
+            Image.asset(
+                icon,
+                height: 60,
+            ),
+            const SizedBox(height: 3),
```
Comment (Denys-Doskochynskyi) [inline lib/components/community_suggestions.dart:40]: I think the better way is to use margin

File: lib/components/article_suggestions.dart [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+  }
+
+  Widget _articleCommunity(BuildContext context, Community community) {
+    return Row(
+      children: [
+        Image.asset(
+          community.icon,
+          height: 15,
+        ),
+        const SizedBox(width: 10),
```
Comment (Denys-Doskochynskyi) [inline lib/components/article_suggestions.dart:76]: I think the better way is to use margin :)

**Review by Denys-Doskochynskyi — APPROVED** [review]
Nice job, nice code and view
Please merge this

File: lib/components/community_suggestions.dart [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+      child: _communityList(context),
+    );
+  }
+
+  Widget _communityList(BuildContext context) {
+    return SingleChildScrollView(
+            scrollDirection: Axis.horizontal,
+            child: Row(
+              children: communities
+                  .map((community) => _communityCard(context, community.name, community.icon))
```
Comment (igor-leshkevych) [inline lib/components/community_suggestions.dart:23]: 👍

**Review by igor-leshkevych — APPROVED** [review]
@aekrops very nice job. The only thing I would change in terms of UI is text alignment of your news block (now text is centered, but I would align it to the left). In terms of code everything is nice and clean :)

+8/8

**Review by lizooo — APPROVED** [review]
Looks really nice! Love your project structure, seems clean and organized (might steal it for my future tasks). Well done in terms of code quality, the only small remark i have wood be paying a bit more attention to the proportions of elements in original application, apart from that everything is superb:) You should totally merge

**Pull request merged**